rows.go

```diff
@@ -197,6 +213,14 @@ func (rows *snowflakeRows) NextResultSet() error {
 	return rows.ChunkDownloader.nextResultSet()
 }
 
+func (rows *snowflakeRows) checkAsyncQueryStatus() {
+	// if async query, block until query is finished
+	if rows.status == QueryStatusInProgress {
+		rows.status = <-rows.statusChannel
+		close(rows.statusChannel)
```

Comment:
In general it's a good idea to close a channel on the writer side, not the reader side. Reading from a closed channel gives you a zero-value but writing to a closed channel panics. It's easier to spot a write on a closed channel if you do the closing around the place where you do the writing.